dask/array/rechunk.py

```diff
@@ -589,7 +619,8 @@ def format_blocks(blocks):
     2*[10] | [5, 6] | 3*[2] | [7]
     """
     assert (isinstance(blocks, tuple) and
-            all(isinstance(x, int) for x in blocks))
+            all(isinstance(x, int) or np.isnan(x)
+                for x in blocks))
```

Comment:
No clue if this is a bottleneck, but `math.isnan` is significantly faster on scalars.